types/src/block.rs

```diff
@@ -400,6 +399,19 @@ impl Block {
         }
     }
 
+    /// Returns a list of all transaction hashes in a block.
+    pub fn all_transaction_hashes(&self) -> Box<dyn Iterator<Item = TransactionHash> + '_> {
+        match self {
+            Block::V1(block) => Box::new(
+                block
+                    .body
+                    .deploy_and_transfer_hashes()
+                    .map(TransactionHash::from),
+            ),
+            Block::V2(block_v2) => Box::new(block_v2.all_transactions().copied()),
+        }
```

Comment:
nitpick, but since we've got the itertools dependency, we could implement this without boxing
```suggestion
    pub fn all_transaction_hashes(&self) -> impl Iterator<Item = TransactionHash> + '_ {
        match self {
            Block::V1(block) => Either::Left(
                block
                    .body
                    .deploy_and_transfer_hashes()
                    .map(TransactionHash::from),
            ),
            Block::V2(block_v2) => Either::Right(block_v2.all_transactions().copied()),
        }
```

Reply:
changed in: https://github.com/casper-network/casper-node/pull/4723/commits/0b8afefd297e3d6e37d3b4404e02d13ba9fa0b87